### PR TITLE
🎉 etlcli: new CLI to rule them all

### PR DESCRIPTION
**Pull request by lucasrodes**
## TL; DR
I've compiled all the various commands that we have, and packed them under a single CLI: `etlcli`. In the future, we may rename this (or add an alias) so it is simply called `etl`.

**Some notes:**
- All commands are now accessible from `etlcli` (see `etlcli --help`).
- I've kept Wizard (`etl-wizard`) separately, as it uses streamlit functionalities that otherwise would be loaded when running `etlcli [any command]`. However, I've added a new alias for it: `etlwiz`, which I believe is shorter and simpler.
- Some commands are grouped in subgroups: (i) "Backport"-related commands (run `etlcli b --help`) and (ii) "Development"-related commands (run `etlcli d --help`).
- Old command names are no longer useful except for:
  - `etl`: I believe I've removed all references to it in the [etl](https://github.com/owid/etl) and the [ops](https://github.com/owid/ops) repos. But will leave if for now.
  - `compare`: The name is too generic, and was hard to ensure that I correctly replaced all references to it.
  - `backport`: Idem as with `etl`.
  - `fasttrack`: We are in the process of deprecating this tool, so I did not migrate it under `etlcli`. Once Wizard replacement is proven to be good enough, I'll be removing it.

**Other minor Wizard stuff**
Now, if one sets the environment variable `WIZARD_IS_REMOTE` to 1, only the apps that are remote-friendly will be rendered in the app.

Also fixed other minor bugs.

### How to review?
I'd just run some trials on `etlcli` and see how it reacts, especially on those commands that are most commonly used (i.e. `etlcli run`, former `etl`).

Forget about reviewing all the 62 files. But if you want to see the entrypoint for the command, you can refer to `apps/cli/__ini__.py`.

### Current commands
(shown as "command name = path to script")

| **alias**             | **path**                                      | **new alias**        | **removed most references** |
|-----------------------|-----------------------------------------------|----------------------|-----------------------------|
| `etl`                   | 'etl.command:main_cli'                        | `etlcli run` (leaving the old alias for some time to avoid issues)         |                        ✅ ❓ (I checked in this repo and https://github.com/owid/ops)         |
| `etl-harmonize`         | 'etl.harmonize:harmonize'                     | `etlcli harmonize`   |          ✅                    |
| `compare`               | 'etl.compare:cli'                             | `etlcli compare` (leaving old alias for some time to avoid issues)     |         ✅ ❓ (too generic word, too many matches on search, I think mostly renamed)                     |
| `etl-datadiff`          | 'etl.datadiff:cli'                            | `etlcli diff`        |                  ✅             |    
| `fasttrack`             | 'apps.fasttrack.cli:cli'                      | (TO BE DEPRECATED)   |                    ✅              |
| `etl-metagpt`           | 'apps.metagpt.cli:main'                       | `etlcli metadata upgrade`      |       ✅                          |
| `etl-wizard`            | 'apps.wizard.cli:cli'                         | `etlwiz`    (leaving the other alias too for some time to avoid issues)      |              ✅                  |
| `etl-staging-sync`      | 'apps.staging_sync.cli:cli'                   | `etlcli chart-sync`       |                  ✅           |
| `version_tracker`       | 'etl.version_tracker:run_version_tracker_checks' |       `etlcli d version-tracker`               |             ✅                 |
| `reindex`               | 'etl.reindex:reindex_cli'                     |           `etlcli d reindex`                 |                  ✅                |
| `publish`               | 'etl.publish:publish_cli'                     |          `etlcli d publish`            |              ✅                    |
| `prune`                 | 'etl.prune:prune_cli'                         |           `etlcli d prune`           |                      ✅        |
| `run_python_step`       | 'etl.run_python_step:main'                    |                `etlcli d run-python-step`            |         ✅                          |
| `generate_graph`        | 'etl.to_graphviz:to_graphviz'                 |           `etlcli graphviz`           |   ✅                           |
| `etl-match-variables`   | 'etl.match_variables:main_cli'                |             `etlcli variable-match`         |         ✅                    |
| `etl-translate-varmap`  | 'etl.variable_mapping_translate:main_cli'     |           `etlcli variable-mapping-translate`           |      ✅                       |
| `etl-chart-suggester`   | 'etl.chart_revision.cli:main_cli'             |          `etlcli chart-upgrade`            |            ✅                 |
| `etl-metadata-export`   | 'etl.metadata_export:cli'                     |         `etlcli metadata-export`             |                   ✅          |
| `etl-metadata-migrate`  | 'apps.metadata_migrate.cli:cli'               |         `etlcli metadata-migrate`             |          ✅                   |
|` etl-chartgpt`          | 'etl.chart_revision.v2.chartgpt:cli'          |       `etlcli chart-revisions`               |                ✅             |
| `etl-staging-sync-app`  | 'apps.staging_sync.app:cli'                   |         REMOVED             |                       ✅         |
| `backport`  | 'apps.backport.backport:backport_cli'                   |        `etlcli b run`             |                       ✅         |
| `backport-migrate`  | 'apps.backport.migrate.migrate:cli'                   |         `etlcli b migrate`             |                       ✅         |
| `bulk_backport`  | 'apps.backport.bulk_backport:bulk_backport'                   |         `etlcli b bulk`             |                       ✅         |
| `backport-fasttrack`  | 'apps.backport.fasttrack_backport:cli'                   |         `etlcli b fasttrack`             |                       ✅         |

### CODE REVIEW

**Comment by lucasrodes** [comment]
TODO:
- Review CI/CD tests
- Style & review `--help`

**Comment by lucasrodes** [comment]
Doing a one-pass review of the `--help` text for all commands:

- [ ] `run`
- [ ] `harmonize`
- [ ] `diff`
- [ ] `graphviz`
- [ ] `compare`
- [ ] `metadata-export`
- [ ] `metadata-migrate`
- [ ] `metadata-upgrade`
- [ ] `chart-sync`
- [ ] `chart-gpt`
- [ ] `chart-revisions`
- [ ] `backport-run`
- [ ] `backport-bulk`
- [ ] `backport-fasttrack`
- [ ] `backport-migrate`
- [ ] `variable-match`
- [ ] `variable-mapping-translate`
- [ ] `dev version-tracker`
- [ ] `dev prune`
- [ ] `dev reindex`
- [ ] `dev publish`
- [ ] `dev run-python-step`

File: .vscode/launch.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
             "type": "python",
```
Comment (Marigold) [inline .vscode/launch.json:9]: Just delete this entry, it shouldn't be here.

File: etl/config.py [hunk 2]
```diff
@@ -138,3 +138,10 @@ def enable_bugsnag() -> None:
         bugsnag.configure(
             api_key=BUGSNAG_API_KEY,
         )  # type: ignore
+
+
+# For Wizard to know if it is running on remote
+## This could eventuallyb be just a generic flag for other apps/processes too.
+## Why is this important for Wizard? Refer to field `disable_on_remote` in apps/wizard/config/config.yml`
+WIZARD_IS_REMOTE = env.get("WIZARD_IS_REMOTE") in ("True", "true", "1")
+print(WIZARD_IS_REMOTE)
```
Comment (Marigold) [inline etl/config.py:147]: This `print` shouldn't be here I assume? (It's printing `False` on local)

File: apps/cli/__init__.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+
+################################
+#
+# RICH CLICK CONFIG
+# Configuration rich_click
+#
+################################
+
+# RICH-CLICK CONFIGURATION
```
Comment (Marigold) [inline apps/cli/__init__.py:185]: This setup is really cool!

File: apps/metadata_migrate/cli.py [hunk 4]
```diff
@@ -63,25 +75,35 @@ def cli(
...
+    ```
+    STAGING=mojmir etlcli metadata-migrate --chart-slug political-regime
+    ```
 
-        # create YAML file in grapher step
-        STAGING=mojmir etl-metadata-migrate --chart-slug political-regime
+    # Reference
     """
-    assert config.STAGING, "You have to run this as STAGING=mystaging etl-metadata-migrate ..."
+    assert config.STAGING, "You have to run this as STAGING=mystaging etlcli-metadata migrate ..."
```
Comment (Marigold) [inline apps/metadata_migrate/cli.py:106]: This should be `etlcli metadata-migrate`, right?

File: apps/wizard/cli.py [hunk 5]
```diff
@@ -18,6 +18,16 @@
 # Disable streamlit cache data API logging
 # ref: @kajarenc from https://github.com/streamlit/streamlit/issues/6620#issuecomment-1564735996
 logging.getLogger("streamlit.runtime.caching.cache_data_api").setLevel(logging.ERROR)
+# RICH-CLICK CONFIGURATION
```
Comment (Marigold) [inline apps/wizard/cli.py:21]: This is repeated in `apps/cli/__init__.py`. Can we put it in one place? (I'd like to reuse it too)

File: etl/chart_revision/cli.py [hunk 6]
```diff
@@ -23,21 +16,48 @@
...
 @click.argument(
     "mapping-file",
     type=str,
 )
-@click.option("--revision-reason", default=None, help="Assign a reason for the suggested chart revision.")
+@click.option(
+    "--revision-reason",
+    "-r",
+    default=None,
+    show_default=True,
```
Comment (Marigold) [inline etl/chart_revision/cli.py:28]: You could also enable it for all commands with
```
@click.command(context_settings=dict(show_default=True))
```

**Review by Marigold — APPROVED** [review]
Looks good! The only thing is that there's now an additional overhead when running commands. For instance adding `returning None` right at the beginning of a function `command/main_cli` and running
```
time etlcli run cherry
```
takes >3s, while `time etl cherry` takes only 1.7s (same for all other commands). Is there a way to make it faster? Perhaps there's a script that preloads something that is responsible for the difference?

Also before we merge this, we should reflect these changes in `ops` and Buildkite jobs (I'll do that myself, but we should coordinate merging).
